dunai/src/Data/MonadicStreamFunction/InternalCore.hs

```diff
@@ -133,3 +133,8 @@ reactimate :: Monad m => MSF m () () -> m ()
 reactimate sf = do
   (_, sf') <- unMSF sf ()
   reactimate sf'
+
+vain :: Monad m => MSF m a b -> MSF m a (b, MSF m a b)
+vain msf0 = MSF { unMSF = f }
```

Comment:
Similar to my comment for `link`, it's maintenance effort to add new functions that use the `MSF` constructor. In this case, it seems pretty fundamental. Can you express any other functions in term of it?

Reply:
You can express parallel switching in terms of `vain`, as in section 4 of the paper.

Reply:
Great, that sounds useful :) Then such combinators should be added.

Reply:
Can that happen separately, or does that need to be in this branch?

Reply:
I personally think it should happen here.

Reply:
In that case I have another branch that implements Tasks with `link` and `vain`, and I could merge it into this branch.